rocky/reports/views/aggregate_report.py

```diff
@@ -116,6 +116,9 @@ def setup(self, request, *args, **kwargs):
         )
 
     def post(self, request, *args, **kwargs):
+        if not hasattr(ReportTypesSelectionAggregateReportView, "selected_oois"):
+            self.setup(request, *args, **kwargs)
+
```

Comment:
This is becoming spaghetti here. `setup` is called by automatically by the framework and always before the `dispatch` method (which in turn calls `post`, `get` or other HTTP methods), so there shouldn't be a call to `setup` here. The `setup` method is responsible for initializing some attributes, including `self.args` and `self.kwargs`. By calling `setup` manually in the `post` method, you're potentially re-initializing or overwriting attributes that the framework has already set up, which could lead to unintended behavior. Since Django guarantees that setup is called before the post method, there's no need to call it again.

---

Also, for a next usage of `hasattr`, the correct way to use `hasattr` in this context is to use: `hasattr(self, '<attribute>')`. Otherwise, what you're doing is checking for a class-level attribute, which is not the intended use and will probably not work, since `selected_oois` hasn't been set in this class, or there were no assignments such as `ReportTypesSelectionAggregateReportView.selected_oois = ...`